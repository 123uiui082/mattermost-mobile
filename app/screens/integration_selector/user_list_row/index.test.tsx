// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import Database from '@nozbe/watermelondb/Database';
import React from 'react';

import {Preferences} from '@app/constants';
import {renderWithEverything} from '@test/intl-test-helper';
import TestHelper from '@test/test_helper';

import UserListRow from '.';

describe('components/integration_selector/user_list_row', () => {
    let database: Database;
    const userProfile: UserProfile = {
        id: '1',
        create_at: 1111,
        update_at: 1111,
        delete_at: 0,
        username: 'johndoe',
        nickname: 'johndoe',
        first_name: 'johndoe',
        last_name: 'johndoe',
        position: 'hacker',
        roles: 'admin',
        locale: 'en_US',
        notify_props: {
            channel: 'true',
            comments: 'never',
            desktop: 'all',
            desktop_sound: 'true',
            email: 'true',
            first_name: 'true',
            mention_keys: 'false',
            push: 'mention',
            push_status: 'ooo',
        },
        email: 'johndoe@me.com',
        auth_service: 'dummy',
    };
    beforeAll(async () => {
        const server = await TestHelper.setupServerDatabase();
        database = server.database;
    });

    it('should render', () => {
        const wrapper = renderWithEverything(
            <UserListRow
                id='1234'
                isMyUser={false}
                theme={Preferences.THEMES.denim}
                user={userProfile}
                teammateNameDisplay='dummy'
                testID='UserListRow'
                enabled={true}
                selectable={false}
                selected={false}
                onPress={() => {
                    // noop
                }}
            >
                <br/>
            </UserListRow>,
            {database},
        );

        expect(wrapper.toJSON()).toBeTruthy();
    });

    it('match snapshot if user is not shared', () => {
        userProfile.remote_id = '1';

        const wrapper = renderWithEverything(
            <UserListRow
                id='1234'
                isMyUser={false}
                theme={Preferences.THEMES.denim}
                user={userProfile}
                teammateNameDisplay='dummy'
                testID='UserListRow'
                enabled={true}
                selectable={false}
                selected={false}
                onPress={() => {
                    // noop
                }}
            >
                <br/>
            </UserListRow>,
            {database},
        );

        expect(wrapper.toJSON()).toMatchSnapshot();
    });

    it('match snapshot if user is my user', () => {
        userProfile.remote_id = '';

        const wrapper = renderWithEverything(
            <UserListRow
                id='1234'
                isMyUser={true}
                theme={Preferences.THEMES.denim}
                user={userProfile}
                teammateNameDisplay='dummy'
                testID='UserListRow'
                enabled={true}
                selectable={false}
                selected={false}
                onPress={() => {
                    // noop
                }}
            >
                <br/>
            </UserListRow>,
            {database},
        );

        expect(wrapper.toJSON()).toMatchSnapshot();
    });
});
