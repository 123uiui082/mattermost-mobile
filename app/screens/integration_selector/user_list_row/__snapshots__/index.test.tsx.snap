// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/integration_selector/user_list_row match snapshot if user is my user 1`] = `
<View
  style={
    {
      "flex": 1,
      "flexDirection": "row",
      "overflow": "hidden",
      "paddingHorizontal": 15,
    }
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "height": 65,
        "opacity": 1,
      }
    }
    testID="UserListRow"
  >
    <View
      style={
        {
          "flex": 1,
          "flexDirection": "row",
        }
      }
      testID="1"
    >
      <View
        style={
          {
            "alignItems": "center",
            "color": "#3f4350",
            "flexDirection": "row",
          }
        }
      >
        <View
          style={
            {
              "borderRadius": 17.5,
              "height": 34,
              "width": 34,
            }
          }
          testID="UserListRow.1.profile_picture"
        >
          <Icon
            name="account-outline"
            size={24}
            style={
              {
                "color": "rgba(63,67,80,0.48)",
              }
            }
          />
        </View>
      </View>
      <View
        style={
          {
            "flex": 1,
            "flexDirection": "column",
            "justifyContent": "center",
            "marginLeft": 10,
          }
        }
        testID="UserListRow.1"
      >
        <View>
          <View
            style={
              {
                "flexDirection": "row",
              }
            }
          >
            <Text
              ellipsizeMode="tail"
              numberOfLines={1}
              style={
                {
                  "color": "#3f4350",
                  "fontSize": 15,
                }
              }
              testID="UserListRow.display_username"
            >
              @johndoe - you
            </Text>
          </View>
        </View>
        <View>
          <Text
            ellipsizeMode="tail"
            numberOfLines={1}
            style={
              {
                "color": "rgba(63,67,80,0.5)",
                "fontSize": 15,
              }
            }
          >
            dummy
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`components/integration_selector/user_list_row match snapshot if user is not shared 1`] = `
<View
  style={
    {
      "flex": 1,
      "flexDirection": "row",
      "overflow": "hidden",
      "paddingHorizontal": 15,
    }
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "height": 65,
        "opacity": 1,
      }
    }
    testID="UserListRow"
  >
    <View
      style={
        {
          "flex": 1,
          "flexDirection": "row",
        }
      }
      testID="1"
    >
      <View
        style={
          {
            "alignItems": "center",
            "color": "#3f4350",
            "flexDirection": "row",
          }
        }
      >
        <View
          style={
            {
              "borderRadius": 17.5,
              "height": 34,
              "width": 34,
            }
          }
          testID="UserListRow.1.profile_picture"
        >
          <Icon
            name="account-outline"
            size={24}
            style={
              {
                "color": "rgba(63,67,80,0.48)",
              }
            }
          />
        </View>
      </View>
      <View
        style={
          {
            "flex": 1,
            "flexDirection": "column",
            "justifyContent": "center",
            "marginLeft": 10,
          }
        }
        testID="UserListRow.1"
      >
        <View>
          <View
            style={
              {
                "flexDirection": "row",
              }
            }
          >
            <Text
              ellipsizeMode="tail"
              numberOfLines={1}
              style={
                {
                  "color": "#3f4350",
                  "fontSize": 15,
                }
              }
              testID="UserListRow.display_username"
            >
              @johndoe
            </Text>
          </View>
        </View>
        <View>
          <Text
            ellipsizeMode="tail"
            numberOfLines={1}
            style={
              {
                "color": "rgba(63,67,80,0.5)",
                "fontSize": 15,
              }
            }
          >
            dummy
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
